scripts/release/publish.mjs

```diff
@@ -0,0 +1,140 @@
+import { execFileSync } from 'child_process';
+import fs from 'fs';
+import { fetch, stream } from 'undici';
+import sade from 'sade';
+
+let DEBUG = false;
+const log = {
+	debug: (...msgs) => DEBUG && console.log(...msgs),
+	info: (...msgs) => console.log(...msgs),
+	error: (...msgs) => console.error(...msgs)
+};
+
+/**
+ * Parse HTTP Link headers into an array of [rel_name, uri] tuples.
+ * This implementation only looks for one parameter named "rel".
+ * https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Link
+ * @param {string} linkHeader
+ * @returns {Array<[string, string]>}
+ */
+function parseLinkHeader(linkHeader) {
+	log.debug('raw link:', linkHeader);
+
+	/** @type {Array<[string, string]>} */
+	const result = [];
+	const uris = linkHeader.split(/,\s*</);
+
+	for (let uri of uris) {
+		// We assume each link uri only has one param and the param key is "rel"
+		// The first "<" may get stripped off by the split call above
+		const match = uri.match(/<?([^>]*)>\s*;\s*rel="?([^"]*)"?/);
+
+		// 1st group: URI, 2nd group: rel name
+		result.push([match[2], match[1]]);
+	}
+
+	log.debug('parsed:', result);
+	return result;
+}
+
+/**
+ * AsyncGenerator for all GitHub releases in this repo, yielding each page of results
+ * @typedef {import('@octokit/openapi-types').components["schemas"]["release"]} Release
+ * @returns {AsyncGenerator<Release[]>}
+ */
+async function* getReleases() {
+	let nextUrl = 'https://api.github.com/repos/preactjs/preact/releases';
+	while (nextUrl) {
+		log.debug('Fetching', nextUrl);
+
+		const response = await fetch(nextUrl);
+		const linkHeader = response.headers.get('Link');
+		const links = linkHeader ? parseLinkHeader(linkHeader) : null;
+
+		nextUrl = links?.find(link => link[0] == 'next')?.[1];
+		yield /** @type {Release[]} */ (await response.json());
+	}
+}
+
+async function main(tag, opts) {
+	DEBUG = opts.debug;
+
+	log.debug('Git tag:', tag);
+	log.debug('Options:', opts);
+
+	// 1. Find a release with the matching tag
+	/** @type {Release} */
+	let release;
+
+	for await (let releasePage of getReleases()) {
```

Comment:
QQ: would it work to use the "get release by tag name" API?
`GET https://api.github.com/repos/preactjs/preact/releases/tags/${tag}`

https://docs.github.com/en/rest/reference/releases#get-a-release-by-tag-name

Reply:
Oh! I didn't use this at first cuz it doesn't return draft releases (and I thought we'd need them here), but since we are running this after publishing the release, it'd be much simpler. Good call!